File: storydoc-frontend/src/app/api/models/block-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ArtifactDto } from './artifact-dto';
import { BlockId } from './block-id';
export interface BlockDto {
  artifacts?: Array<ArtifactDto>;
  blockId?: BlockId;
  blockType?: string;
  numbering?: Array<number>;
  parentBlockId?: BlockId;
  text?: string;
  title?: string;
}

File: storydoc-backend/src/main/java/io/storydoc/server/storydoc/app/dto/BlockDTO.java
package io.storydoc.server.storydoc.app.dto;

import io.storydoc.server.storydoc.domain.BlockId;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BlockDTO {

    private BlockId blockId;

    private int[] numbering;

    private BlockId parentBlockId;

    public String blockType;

    private String title;

    private String text;

    private List<ArtifactDTO> artifacts;

}
